src/cryptography/x509/name.py

```diff
@@ -138,6 +164,23 @@ def rfc4514_string(self) -> str:
             _escape_dn_value(self.value),
         )
 
+    @classmethod
+    def from_rfc4514_string(cls, value: str) -> "NameAttribute":
+        """
+        Parse a RFC4514 formatted Distinguished Name string.
+        """
+        attr_type, attr_value = value.split("=", 1)
+
+        # if a descr, it's case insensitive (RFC 4512, section 1.4)
+        attr_type = attr_type.upper()
+        attr_value = _parse_dn_value(attr_value)
+
+        if attr_type in _NAME_TO_NAMEOID:
+            return NameAttribute(_NAME_TO_NAMEOID[attr_type], attr_value)
+        elif re.match(r"[0-9]+(\.[0-9]+)*$", attr_type):
```

Comment:
All regexps should be made into constants with `re.compile`.

Reply:
agreed, fixed.